scrcpy/src/main/jni/adb_pairing.cpp

```
@@ -0,0 +1,302 @@
+    );
+    if (status != 1 || key_size == 0) {
+        LOGE("Unable to generate the SPAKE2 public key.");
+
+        env->ReleaseByteArrayElements(jPassword, pswd, 0);
+        SPAKE2_CTX_free(spake2_ctx);
+        return 0;
+    }
+    env->ReleaseByteArrayElements(jPassword, pswd, 0);
+
+    auto ctx = reinterpret_cast<PairingContextNative*>(malloc(sizeof(PairingContextNative)));
+    memset(ctx, 0, sizeof(PairingContextNative));
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)